hedera-node/src/main/java/com/hedera/services/ledger/HederaLedger.java

```diff
@@ -454,6 +464,13 @@ public boolean isDeleted(AccountID id) {
 		return (boolean) accountsLedger.get(id, IS_DELETED);
 	}
 
+	public boolean isDetached(AccountID id) {
+		return dynamicProperties.autoRenewEnabled()
+				&& !(boolean) accountsLedger.get(id, IS_SMART_CONTRACT)
+				&& (long) accountsLedger.get(id, BALANCE) == 0L
+				&& !validator.isAfterConsensusSecond((long) accountsLedger.get(id, EXPIRY));
```

Comment:
I see this function is only used with negation. To increase readability, perhaps we should rename it to `isWithinConsensusSecond` and make corresponding changes.